### PR TITLE
Add per particle listener support to CoulombPBCAB

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes

This contains changes to CoulombPBCAB to support per particle Listeners and a corresponding test. Documentation is partial here but should make sense when the changes to QMCHamiltonian are PR'd. That can't really be done until all the lower level changes go in.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature
- Code style update (formatting, renaming)
- Build related changes
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Utilities/ResourceCollection.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <string>
 #include <memory>
 #include <cstddef>
+#include <cassert>
```
Comment (ye-luo) [inline src/Utilities/ResourceCollection.h:18]: Is this the correct place? Don't feel right adding such include just for convenience.

Comment (PDoakORNL) [reply src/Utilities/ResourceCollection.h]: It was the wrong place.  Fixed now.

**Review by ye-luo — DISMISSED** [review]

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**